fabric-client-gametest-api-v1/src/client/java/net/fabricmc/fabric/api/client/gametest/v1/ClientGameTestInput.java

```diff
@@ -0,0 +1,318 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.client.gametest.v1;
+
+import java.util.function.Function;
+
+import org.jetbrains.annotations.ApiStatus;
+
+import net.minecraft.client.gui.screen.Screen;
+import net.minecraft.client.option.GameOptions;
+import net.minecraft.client.option.KeyBinding;
+import net.minecraft.client.util.InputUtil;
+
+/**
+ * The client gametest input handler used to simulate inputs to the client.
+ */
+@ApiStatus.NonExtendable
+public interface ClientGameTestInput {
+	/**
+	 * Starts holding down a key binding. The key binding will be held until it is released. The key binding must be
+	 * bound. Does nothing if the key binding is already being held.
+	 *
+	 * @param keyBinding The key binding to hold
+	 * @see #releaseKey(KeyBinding)
+	 * @see #pressKey(KeyBinding)
+	 * @see #holdKey(Function)
+	 */
+	void holdKey(KeyBinding keyBinding);
+
+	/**
+	 * Starts holding down a key binding. The key binding will be held until it is released. The key binding must be
+	 * bound. Does nothing if the key binding is already being held.
+	 *
+	 * @param keyBindingGetter The function to get the key binding from the game options
+	 * @see #releaseKey(Function)
+	 * @see #pressKey(Function)
+	 * @see #holdKey(KeyBinding)
+	 */
+	void holdKey(Function<GameOptions, KeyBinding> keyBindingGetter);
+
+	/**
+	 * Starts holding down a key or mouse button. The key will be held until it is released. Does nothing if the key or
+	 * mouse button is already being held.
+	 *
+	 * @param key The key or mouse button to hold
+	 * @see #releaseKey(InputUtil.Key)
+	 * @see #pressKey(InputUtil.Key)
+	 */
+	void holdKey(InputUtil.Key key);
+
+	/**
+	 * Starts holding down a key. The key will be held until it is released. Does nothing if the key is already being
+	 * held.
+	 *
+	 * @param keyCode The key code of the key to hold
+	 * @see #releaseKey(int)
+	 * @see #pressKey(int)
+	 */
+	void holdKey(int keyCode);
```

Comment:
What does keyCode mean here? Is it one of the GLFW ones? `@MagicConstant(valuesFromClass = GLFW.class)` may help here.

dito for the other methods here.

Reply:
Yeah, it's the GLFW ones, but aren't people supposed to prefer the constants in `InputUtil`, and only if it's not there use the ones in `GLFW`? Or at least, I don't want to generate warnings for the people who use the ones in `InputConstants`. There are also extra constants in both which are not applicable. Same for the mouse methods.

Reply:
I put the fact that it's a GLFW key code in the javadocs, at least.

Reply:
Yes, that would be good, I wasnt aware of InputUtil.